internal/observability/exporter/prometheus/prometheus_option.go

```diff
@@ -0,0 +1,35 @@
+//
+// Copyright (C) 2019-2020 Vdaas.org Vald team ( kpango, rinx, kmrmt )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package prometheus provides a prometheus exporter.
+package prometheus
+
+type PrometheusOption func(*prometheusOptions) error
+
+var (
+	prometheusDefaultOpts = []PrometheusOption{
+		WithNamespace("vald"),
```

Comment:
in other component (eg. gateway, indexer ) uses default as namespace, should we concider about vald global default namespace setting?

Reply:
this is not a option to specify k8s namespace. it's just a prefix string for the metrics exported to Prometheus.

However, as you pointed out, it is good to consider "vald" as global default k8s namespace. let's discuss it in another issue.